mx.py

```diff
@@ -1713,15 +1716,21 @@ def _getCompiler(self):
         if self.args.jdt and not self.args.force_javac:
             return ECJCompiler(self.args.jdt, self.args.extra_javac_args)
         else:
-            return JavacCompiler(self.args.alt_javac, self.args.extra_javac_args)
+            if self.args.no_javac_daemon:
+                return JavacCompiler(self.args.alt_javac, self.args.extra_javac_args)
+            else:
+                return JavacCompilerDaemon(self.args.alt_javac, self.args.extra_javac_args)
```

Comment:
Maybe we should abort if `alt_javac` is set with a message saying that `--no-javac-daemon` needs to be set when using `--alt-javac`.
Alternatively we can set that up automatically after parsing arguments in `build` by setting `no_javac_daemon` to true if `alt_javac` is true (with a warning maybe)